### ISSUE TITLE
Terraform deletes my ACR registry url reference in Function App when doing any kind of update to it

### ISSUE THREAD

**Issue by joovieira123** [hashicorp/terraform-provider-azurerm#20983]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

v1.3.7

### AzureRM Provider Version

v3.47.0

### Affected Resource(s)/Data Source(s)

azurerm_linux_function_app

### Terraform Configuration Files

```hcl
resource "azurerm_linux_function_app" "staging_function_app" {
  name                = "wf-staging-afc"
  location            = azurerm_resource_group.worldfavor_staging.location
  resource_group_name = azurerm_resource_group.worldfavor_staging.name

  service_plan_id             = data.azurerm_service_plan.function_app_service_plan.id
  storage_account_name        = data.azurerm_storage_account.function_app_storage_account.name
  storage_account_access_key  = data.azurerm_storage_account.function_app_storage_account.primary_access_key
  https_only                  = true
  functions_extension_version = "~4"
  client_certificate_enabled  = true
  client_certificate_mode     = "Required"

  app_settings = merge(
    {
      FUNCTION_APP_EDIT_MODE              = "readOnly"
      WEBSITES_ENABLE_APP_SERVICE_STORAGE = false
      DOCKER_ENABLE_CI                    = true
      https_only                          = true
      ResponseSubmittedSubscriptionName   = "operation-countries"
      ResponseSubmittedTopicName          = "response-submitted"
      FileServiceUrl                      = "https://test.services.worldfavor.com/file-management/file/"
      ExportServiceUrl                    = "https://test.services.worldfavor.com/file-management/export/"
    },
    {
      for secret_key, secret_value in local.keyvault_secrets :
      secret_value => "@Microsoft.KeyVault(SecretUri=${azurerm_key_vault.wf_kv_staging.vault_uri}secrets/${data.azurerm_key_vault_secret.staging_keyvault_secret_name[secret_key].name})"
    }
  )

  site_config {
    always_on                              = true
    ftps_state                             = "FtpsOnly"
    http2_enabled                          = true
    use_32_bit_worker                      = false
    application_insights_connection_string = data.azurerm_application_insights.dev_appinsights.connection_string
    application_insights_key               = data.azurerm_application_insights.dev_appinsights.instrumentation_key
    cors {
      allowed_origins     = []
      support_credentials = false
    }
    application_stack {
      docker {
        registry_url      = "worldfavorregistry.azurecr.io"
        image_name        = "azure-functions"
        image_tag         = "23048"
        registry_username = data.azurerm_container_registry.worldfavor_acr.admin_username
        registry_password = "@Microsoft.KeyVault(SecretUri=${data.azurerm_key_vault.internal_kv.vault_uri}secrets/${data.azurerm_key_vault_secret.docker_registry_server_password.name}/${data.azurerm_key_vault_secret.docker_registry_server_password.version})"
      }
    }
  }

  identity {
    type = "SystemAssigned"
  }

  lifecycle {
    ignore_changes = [
      tags["hidden-link: /app-insights-instrumentation-key"],
      tags["hidden-link: /app-insights-resource-id"],
      tags["hidden-link: /app-insights-conn-string"],
      site_config[0].cors,
      site_config[0].application_stack[0].docker["image_tag"]
    ]
  }

  tags = local.generic_tags
}
```

### Debug Output/Panic Output

```shell
Can't really see anything in the output because both plan and apply don't show any changes in the docker block above, although the change is happening for sure since I can see it if I run the command az functionapp config show -n wf-staging-afc -g Worldfavor-Staging | jq .linuxFxVersion
```

### Expected Behaviour

Does not override the registry url that the azure function uses to retrieve the image from the registry.
This is the output of the image tag before running a tf apply on anything that touches the function itself and also the output that I would expect after running a tf apply on anything that touches the azure function (unless ofc I was trying to change the registry url itself):

```
az functionapp config show -n wf-staging-afc -g Worldfavor-Staging | jq .linuxFxVersion
/opt/homebrew/Cellar/azure-cli/2.45.0/libexec/lib/python3.10/site-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated
  "class": algorithms.Blowfish,
"DOCKER|worldfavorregistry.azurecr.io/azure-functions:23256"
```

### Actual Behaviour

This is the actual output of the tag after running tf apply on anything that touches the azure function, essentially rendering my azure function useless cause the registry url that the function uses to retrieve the tag from mysteriously went missing:

```
az functionapp config show -n wf-staging-afc -g Worldfavor-Staging | jq .linuxFxVersion
/opt/homebrew/Cellar/azure-cli/2.45.0/libexec/lib/python3.10/site-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated
  "class": algorithms.Blowfish,
"DOCKER|/azure-functions:23256"
```

### Steps to Reproduce

1. In my azure function deploy pipeline I re-deploy the image and then I get the full tag with the registry URL as it's suppose to be.
2. Then I run the command `az functionapp config show -n wf-staging-afc -g Worldfavor-Staging | jq .linuxFxVersion` to verify that the tag is complete.
3. Change anything in the function app...could be something as simple as adding a new dummy app_settings config.
4. Run a `tf apply` on it.
5. Then run the command `az functionapp config show -n wf-staging-afc -g Worldfavor-Staging | jq .linuxFxVersion` again and unfortunately I verify that the registry url part of the docker image tag is imcomplete.

### References

Issue #16991 was very similar to this although it seems like it was closed without any solution.

**Comment by xiaxyi** [comment]
Thanks @joovieira123 for raising this issue, I see this issue is relate to the linuxFxVersion. Currently, we are making some changes to the `linuxFxVersion` and you may refer to the pr https://github.com/hashicorp/terraform-provider-azurerm/pull/20735 for more information.

**Comment by joovieira123** [comment]
Hi @xiaxyi, thank you for letting me know this is something that you're working on. Shall we leave the issue open until the PR is merged?

**Comment by peruzzof** [comment]
I am facing the same issue.

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #22003 which superseded #20735  . As such, I am going to mark this issue as closed.
